### PR TITLE
Presentation code. 

### PR DESCRIPTION
**Pull request by WoozyDragon**
Once the pres hotfixes is merged, we will have software that is tested on the robot and works.

### CODE REVIEW

File: lunabot_control/scripts/manual_controller.py [hunk 1]
```diff
@@ -1,101 +1,187 @@
...
-.buttons
+.buttons   (1 is pressed, 0 not)
 0 A
 1 B
 2 X
 3 Y
 4 LB
 5 RB
-6 back
+6 back (view button)
```
Comment (ashviniyer21) [inline lunabot_control/scripts/manual_controller.py:18]: Wdym view button?

Comment (Isaac-F24) [reply lunabot_control/scripts/manual_controller.py:18]: image

Comment (ashviniyer21) [reply lunabot_control/scripts/manual_controller.py:18]: Oh is that just the name of the button?

File: lunabot_embedded/firmware/teensy_main/lib/lunabot_drivers/RobotMsgs.pb.h [hunk 2]
```diff
@@ -43,13 +39,13 @@ extern "C" {
 
 /* Initializer values for message structs */
 #define RobotSensors_init_default                                                                  \
-  { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }
+  { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 }
```
[the remark this answers is not in the document]
Comment (WoozyDragon) [reply lunabot_embedded/firmware/teensy_main/lib/lunabot_drivers/RobotMsgs.pb.h:42]: going to leave b/c it works for now

File: lunabot_embedded/firmware/teensy_main/lib/lunabot_drivers/interfaces.cpp [hunk 3]
```diff
@@ -252,3 +241,100 @@ void M5Stack_UWB_Trncvr::transfer() {
...
+      return;
+  }
+}
+
+//exc, dep, drive_L, drive_R
+volatile int KillSwitchRelay::cutoff_buffer[4] = {0};
+volatile int KillSwitchRelay::disable_counter[4] = {0};
+volatile bool KillSwitchRelay::is_disable[4] = {false};
+
+void KillSwitchRelay::logic(RobotEffort &effort) {
```
[the remark this answers is not in the document]
Comment (WoozyDragon) [reply lunabot_embedded/firmware/teensy_main/lib/lunabot_drivers/interfaces.cpp:291]: See line 118 - 140 in interfaces.hpp

File: lunabot_control/scripts/manual_controller.py [hunk 4]
```diff
@@ -1,101 +1,187 @@
...
+
 from sensor_msgs.msg import Joy
+from lunabot_msgs.msg import RobotEffort
+
+from enum import Enum
+import numpy as np
 
 """
-.buttons
+.buttons   (1 is pressed, 0 not)
```
[the remark this answers is not in the document]
Comment (Isaac-F24) [reply lunabot_control/scripts/manual_controller.py:11]: I say keep them at least for the value when/not pressed

File: lunabot_control/scripts/manual_controller.py [hunk 5]
```diff
@@ -1,101 +1,187 @@
...
+- Right stick: right wheels
+- Right trigger Excavation forwards (pick up dirt)
+- Left trigger: Excavation backwards
+- D-pad up/down: Linear actuator control (move excavation system up/down)
+- Y button: Latch/unlatch excavation speed
+    - When latched, the excavation speed will stay at its last speed until unlatched
+- X button: Switch between forwards and backwards driving
+    - Forwards is defined as leading with excavation
+- B button: Deposition (spin auger)
+- Start button: Stop the robot
```
[the remark this answers is not in the document]
Comment (Isaac-F24) [reply lunabot_control/scripts/manual_controller.py]: Changed

**Review by ashviniyer21 — COMMENTED** [review]
Thanks for changes. Assuming everything's already been tested looks good to merge. I would like @raghavauppuluri13 to take a look before merging if possible (just to make sure all firmware stuff is good, I'm not that confident in reviewing that)

File: lunabot_embedded/firmware/teensy_main/lib/lunabot_drivers/RobotMsgs.pb.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (raghavauppuluri13) [reply lunabot_embedded/firmware/teensy_main/lib/lunabot_drivers/RobotMsgs.pb.h:42]: @ashviniyer21 this is autogenerated code

**Review by raghavauppuluri13 — APPROVED** [review]

**Comment by raghavauppuluri13** [comment]
LGTM. Just a note that sensors and killswitch logic hasn’t bee tested.

**Pull request merged**